### ISSUE TITLE
Add a way to enable database transactions in PrismaClient

### ISSUE THREAD

**Issue by cannikin** [redwoodjs/redwood#286]
See the workaround that the covid19 devs came up with: https://github.com/lachlanjc/covid19/blob/master/redwood-hack/api-dbInstance.js

Maybe we just designate an ENV var that's false in `.env.defaults` but can be enabled  by a user in `.env`?

@peterp I don't mind adding this to the code real quick, I just want to verify with you that dotenv will be around and will make .env and .env.defaults available at this point in the process? https://github.com/redwoodjs/redwood/blob/master/packages/api/src/dbInstance.ts#L3

**Comment by peterp** [comment]
@cannikin That seems like a good initial solution, I wanted to provide a mechanism that could be used instead of our `dbInstance` in `@redwoodjs/api` because there may be more configuration options that we want to change.

I'm not 100% sure how that would work; but maybe the babel config could check if a file exists, maybe in `./api/src/lib/overwriteDbInstance.js|ts` and use that instead of over here: 
https://github.com/redwoodjs/redwood/blob/dcab7e46a830f6e34fc71e851164366c62223d42/packages/core/config/babel-preset.js#L72-L74 

To summarize:

1. Use getPaths and `fs` to determine if a file exists (`./api/src/lib/overwriteDbInstance.js|ts`)
2. Use that instead of '@redwoodjs/api/dist/dbInstance'

**Comment by cannikin** [comment]
Ahhh interesting! That feels very similar to Rails' Initializers, which are just plain Ruby files you put in a directory `config/initializers` and all of them are run as your app starts up. It's a general place you put all code that requires an init step.

I'd go so far as to suggest we include this file in there by default to make it really obvious to people that it can be changed. Otherwise they'd need to track down some docs somewhere that tell them how to do it. I'll make a PR with a first pass and we can discuss over there!

**Comment by thedavidprice** [comment]
👍 thinking about a similar path on the Jest config side of things. And agree about having the file there by default. Does add to file overhead, though.
